implementations/micrometer-registry-dynatrace/src/test/java/io/micrometer/dynatrace/DynatraceMeterRegistryTest.java

```diff
@@ -151,6 +153,142 @@ void shouldNotTrackPercentilesWithDynatraceSummary() throws Throwable {
+
+                            // DistributionSummary lines
+                            "my.ds,dt.metrics.source=micrometer gauge,min=100,max=100,sum=100,count=1 "
+                                    + clock.wallTime(),
+                            // DistributionSummary percentile lines. Percentiles are 0
+                            // because the step rolled over.
+                            "my.ds.percentile,dt.metrics.source=micrometer,phi=0.5 gauge,0 " + clock.wallTime(),
+                            "my.ds.percentile,dt.metrics.source=micrometer,phi=0.7 gauge,0 " + clock.wallTime(),
+                            "my.ds.percentile,dt.metrics.source=micrometer,phi=0.99 gauge,0 " + clock.wallTime(),
+
+                            // LongTaskTimer lines
+                            "my.ltt,dt.metrics.source=micrometer gauge,min=100,max=100,sum=100,count=1 "
+                                    + clock.wallTime(),
+                            "#my.ltt gauge dt.meta.unit=milliseconds",
+                            // LongTaskTimer percentile lines
+                            // 0th percentile is missing because it doesn't clear the
+                            // "interpolatable line" threshold defined in
+                            // DefaultLongTaskTimer#takeSnapshot().
+                            "my.ltt.percentile,dt.metrics.source=micrometer,phi=0.5 gauge,100 " + clock.wallTime(),
+                            "my.ltt.percentile,dt.metrics.source=micrometer,phi=0.7 gauge,100 " + clock.wallTime(),
+                            "my.ltt.percentile,dt.metrics.source=micrometer,phi=0.99 gauge,100 " + clock.wallTime(),
+                            "#my.ltt.percentile gauge dt.meta.unit=milliseconds"))));
+    }
+
+    @Test
+    void shouldTrackPercentilesWhenDynatraceSummaryInstrumentsNotUsed_shouldExport0PercentileWhenSpecified()
+            throws Throwable {
+        DynatraceConfig dynatraceConfig = getNonSummaryInstrumentsConfig();
+
+        DynatraceMeterRegistry registry = DynatraceMeterRegistry.builder(dynatraceConfig)
+            .httpClient(httpClient)
+            .clock(clock)
+            .build();
+
+        HttpSender.Request.Builder builder = HttpSender.Request.build(config.uri(), httpClient);
+        when(httpClient.post(config.uri())).thenReturn(builder);
+
+        // create instruments with an explicit 0 percentile. This should be exported.
+        Timer timer = Timer.builder("my.timer").publishPercentiles(0, 0.5, 0.99).register(registry);
+        DistributionSummary distributionSummary = DistributionSummary.builder("my.ds")
+            .publishPercentiles(0, 0.5, 0.99)
+            .register(registry);
+        // For LongTaskTimer, the 0 percentile is not tracked as it doesn't clear the
+        // "interpolatable line" threshold defined in DefaultLongTaskTimer#takeSnapshot().
+        // see shouldTrackPercentilesWhenDynatraceSummaryInstrumentsNotUsed for a test
+        // that exports LongTaskTimer percentiles
```

Comment:
This comment is referring to these three lines: https://github.com/micrometer-metrics/micrometer/blob/e3329b8f49c4e0eeec6fc24b10edd9f3c16af040/micrometer-core/src/main/java/io/micrometer/core/instrument/internal/DefaultLongTaskTimer.java#L143-L145
They drop the 0 percentile (`percentilesAboveInterpolatableLine` won't contain the `0` percentile anymore)

I'm not sure if this is a bug or working as intended.

Reply:
I dug a little deeper, and I think the problem lies in the fact that the 0% HistogramGauge is never registered.
1. The LongTaskTimer is registered, with the explicit zero percentile.
2. The histogram gauges get registered. To do so, an initial snapshot is created: 
https://github.com/micrometer-metrics/micrometer/blob/e3329b8f49c4e0eeec6fc24b10edd9f3c16af040/micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/HistogramGauges.java#L100
3. At this point, the number of active tasks is 0. The LongTaskTimer is currently in the process of being registered. Thus, the calculation for `percentilesAboveInterpolatableLine` looks as follows:
https://github.com/micrometer-metrics/micrometer/blob/e3329b8f49c4e0eeec6fc24b10edd9f3c16af040/micrometer-core/src/main/java/io/micrometer/core/instrument/internal/DefaultLongTaskTimer.java#L143-L145
Since the number of `activeTasks` is 0, the calculation in the `filter` effectively evaluates to `p * (0 + 1) > 0`, which is equivalent to `p > 0`. This, of course, is a condition that the 0 percentile (where `p==0`) explicitly does not fulfil. The 0 percentile is dropped here.
4. Because the 0 percentile was dropped in the initial `takeSnapshot`, it will not be registered as a HistogramGauge because the `percentileValues` call has only percentiles > 0: https://github.com/micrometer-metrics/micrometer/blob/e3329b8f49c4e0eeec6fc24b10edd9f3c16af040/micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/HistogramGauges.java#L103
5. Since the 0th percentile gauge is never registered, it is also never exported. 

Later, when the app is running and there are active tasks, this is caught in the `if` here: https://github.com/micrometer-metrics/micrometer/blob/e3329b8f49c4e0eeec6fc24b10edd9f3c16af040/micrometer-core/src/main/java/io/micrometer/core/instrument/internal/DefaultLongTaskTimer.java#L151
However, on the initial `takeSnapshot`, https://github.com/micrometer-metrics/micrometer/blob/e3329b8f49c4e0eeec6fc24b10edd9f3c16af040/micrometer-core/src/main/java/io/micrometer/core/instrument/internal/DefaultLongTaskTimer.java#L161-L165 evaluates to an empty list, as there are no active tasks during registration, and the 0th percentile is never registered.

Reply:
We do have at least one bug in this area of the code. See https://github.com/micrometer-metrics/micrometer/issues/3877. I'll take a closer look tomorrow.

Reply:
Just wanted to update here that I didn't get a chance to look into it more today due to some other work that needs to get done first. I'll come back to this once that's done.

Reply:
It'll take some more time to look into the LTT issue, I think. But the changes in this pull request don't change the status quo, I think. So nothing is worse related to the bug. I think we're fine to merge these changes and follow-up on #3877 separately. Let me know if I'm missing anything.

Reply:
That's right, there is no change in the status quo, it only fixes a bug where the 0 percent quantile wasn't added automatically when explicitly specifying percentiles.